ckanext/security/controllers.py

```diff
@@ -94,8 +95,18 @@ def login(self):
             login = identity['login']
             user = model.User.by_name(login)
 
-            if user is None or not user.is_active() or not user.validate_password(identity['password']):
-                log.info('user failed to provide correct credentials')
+            throttle = LoginThrottle(user, login)
+            locked_out = throttle.is_locked()
+            if locked_out:
+                log.info('User {} attempted login while brute force lockout in place'.format(login))
+
+            invalid_login = user is None or not user.is_active() or not user.validate_password(identity['password'])
+            if invalid_login:
+                # Increment the throttle counter if the login failed.
+                throttle.increment()
```

Comment:
Only increment if the login failed for usual reasons, not if already locked. This matches the existing pattern in the CKANLoginThrottle